homework-submission/myapp4/src/Components/Form.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { inject, observer } from "mobx-react";
+
+@inject("TodosStore")
+@observer
+
+class Form extends React.Component {
+
+  handleSubmit = event => {
+    event.preventDefault();
+
+    const descriptionInput = this.descriptionInput.value;
+    const deadlineInput = this.deadlineInput.value;
+
+    this.props.TodosStore.createTodo({
+      description: descriptionInput,
+      deadline: deadlineInput,
+      __v: 0, done: false
+    });
+    this.descriptionInput.value = "";
+    this.deadlineInput.value = "";
+  };
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
```

Comment:
You have a bug, where you are submitting this form it appears like you are submitting an empty items. but on refresh it shows up as the correct input.

So perhaps you may have another look if you have to re-`addTodos` again on submit 😉